### PR TITLE
feat(web-devtools): custom-context-input

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
- update dispute template presets
- Add option to provide custom context

---

## PR-Codex overview
This PR introduces enhancements to the `DisputeTemplateView` by adding a new component `CustomContextInputs`, which allows for additional context inputs in the dispute process. It also updates some transaction hashes and improves the existing component structure.

### Detailed summary
- Updated `txnHash` values in `FetchDisputeRequestInput.tsx`.
- Changed `title` from "Trump-Biden" to "Trump-Harris" in `FetchDisputeRequestInput.tsx`.
- Added `className` prop to `Container` in `JSONEditor.tsx`.
- Introduced `CustomContextInputs` component to manage additional context inputs.
- Added state for `customContext` in `DisputeTemplateView`.
- Modified initial context setup to include `customContext`.
- Updated dependencies in `DisputeTemplateView` to include `customContext`.
- Integrated `CustomContextInputs` in the form of `DisputeTemplateView`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Release Notes

- **New Features**
	- Introduced a new component, `CustomContextInputs`, for managing additional context variables.
	- Users can now input custom context data in the `DisputeTemplateView`.
	- Updated the placeholder for the `_templateUri` input field for clarity.

- **Enhancements**
	- The `JSONEditor` component now supports custom class names for improved styling.

- **Bug Fixes**
	- Updated `txnHash` values and titles for several presets in the dispute template.

- **Chores**
	- Removed a path mapping entry in the TypeScript configuration to streamline module resolution.

### CODE REVIEW

File: web-devtools/src/app/(main)/dispute-template/page.tsx [hunk 1]
```diff
@@ -280,6 +294,16 @@ const DisputeTemplateView = () => {
...
+          <StyledRow>
+            <StyledP>{"Custom Context :"}</StyledP>
+            <StyledJSONEditor
+              content={{ text: customContextInput }}
+              mode={Mode.text}
+              onChange={(val: any) => {
+                setCustomContextInput(val.text);
+              }}
+            />
+          </StyledRow>
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web-devtools/src/app/(main)/dispute-template/page.tsx]: It handles validation itself, and even shows Repair button to repair the JSON.

File: web-devtools/src/app/(main)/dispute-template/page.tsx [hunk 2]
```diff
@@ -178,7 +184,13 @@
         setLoading(true);
 
         setTimeout(() => {
-          const initialContext = {
+          let customContext = null;
+          try {
+            customContext = JSON.parse(customContextInput);
+          } catch (error) {
+            console.log("Error parsing custom context", error);
+          }
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web-devtools/src/app/(main)/dispute-template/page.tsx]: Not required atm, will add some UI to display errors separately later.

File: web-devtools/src/app/(main)/dispute-template/CustomContextInputs.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const requiredVariables = useMemo(() => retrieveVariables(dataMapping), [dataMapping]);
+
+  useDebounce(
+    () => {
+      if (!customContext) return;
+      setCustomContext(customContext);
+    },
+    300,
+    [customContext]
+  );
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web-devtools/src/app/(main)/dispute-template/CustomContextInputs.tsx:55]: Not required.

File: web-devtools/src/app/(main)/dispute-template/CustomContextInputs.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  "arbitratorDisputeID",
+  "externalDisputeID",
+  "templateID",
+  "templateUri",
+];
+
+interface ICustomContextInputs {
+  dataMapping: string;
+  setCustomContext: (context: Record<string, string>) => void;
+}
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web-devtools/src/app/(main)/dispute-template/CustomContextInputs.tsx:42]: It's just a set of variables to not include in the Custom input as they are already provided in the component before this one. The type is not necessary.

JSDoc for react component is unnecessary here.

**Review by alcercu — APPROVED** [review]
lgtm

**Review by kemuru — APPROVED** [review]

**Pull request merged**